packages/cloud/connection_details/kibana/kibana_connection_details_provider.tsx

```diff
@@ -64,7 +65,9 @@ const createOpts = async (props: KibanaConnectionDetailsProviderProps) => {
           },
         };
       },
-      hasPermission: async () => true,
+      hasPermission: once(
+        async () => await http!.get<boolean>('/internal/security/api_key/check_permissions')
+      ),
```

Comment:
Calling, from a plugin, an HTTP endpoint owned by another plugin is a bad isolation pattern that is strongly discouraged. 

In theory, that should be done by leveraging the security plugin's contract (that should be the one charge of performing the http call). 

Let's wait on what the security team thinks about it.

Reply:
Good call, @pgayvallet . I agree it would be best to expose this alongside the rest of the apiKeys functions in Security plugin's start() > authc > apiKeys